Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -1188,13 +1191,31 @@ namespace Babylon
 
             static Napi::Promise CreateAsync(const Napi::CallbackInfo& info)
             {
-                auto jsSession = info.Env().Global().Get(JS_CLASS_NAME).As<Napi::Function>().New({info[0]});
-                auto& session = *XRSession::Unwrap(jsSession);
-
-                session.m_xr.BeginSession(info.Env());
+                auto jsSession = Napi::Persistent(info.Env().Global().Get(JS_CLASS_NAME).As<Napi::Function>().New({info[0]}));
+                auto& session = *XRSession::Unwrap(jsSession.Value());
 
                 auto deferred = Napi::Promise::Deferred::New(info.Env());
-                deferred.Resolve(jsSession);
+                session.m_xr.BeginSession(info.Env())
+                    .then(session.m_runtimeScheduler,
+                        arcana::cancellation::none(),
+                        [deferred, jsSession = std::move(jsSession), env = info.Env()](const arcana::expected<void, std::exception_ptr>& result) {
+                            if (result.has_error())
```

Comment:
Nit: The indentation/formatting here seems weird. Seems like it would be more readable to have something like:
```c++
session.m_xr.BeginSession(info.Env()).then(session.m_runtimeScheduler, arcana::cancellation::none(),
    [deferred, jsSession = std::move(jsSession), env = info.Env()](const arcana::expected<void, std::exception_ptr>& result) {
        if (result.has_error())
```